op-challenger: Load kzg point evaluations to PreimageOracle

Inphi opened the pull request:
Updates the op-challenger to handle KZG point evaluation preimage keys.

Inphi committed:
op-challenger: load KZG point evaluation into oracle

ajsutton commented on op-e2e/faultproofs/precompile_test.go:
```diff
@@ -0,0 +1,183 @@
+		{
+			name:    "ecrecover",
+			address: common.BytesToAddress([]byte{0x01}),
+			input:   common.Hex2Bytes("18c547e4f7b0f325ad1e56f57e26c745b09a3e503d86e00e5255ff7f715d3d1c000000000000000000000000000000000000000000000000000000000000001c73b1693892219d736caba55bdb67216e485557ea6b6af75f37096c9aa6a5a75feeb940b1d03b21e36b0e47e79769f095fe2ab855bd91e3a38756b7d75a9c4549"),
```
We should changes these to use `common.FromHex` - `Hex2Bytes` is a major foot-gun since it just returns an empty byte array if there's an `0x` prefix.  Semgrep now bans its use.

ajsutton commented on op-e2e/faultproofs/precompile_test.go:
```diff
@@ -0,0 +1,183 @@
+	t.Logf("Completed in %d steps", state.Step)
+}
+
+func waitForSafeHead(ctx context.Context, safeBlockNum uint64, rollupClient *sources.RollupClient) error {
```
Can we make this a reusable function in the `wait` package like we have `wait.ForUnsafeBlock`? https://github.com/ethereum-optimism/optimism/blob/02987acc48daa480a085ac748062a760723b9e2d/op-e2e/e2eutils/wait/blocks.go#L75

Inphi replied:
Done. til of wait/blocks.go. Very useful.

ajsutton commented on op-e2e/faultproofs/util.go:
```diff
@@ -31,3 +43,12 @@ func startFaultDisputeSystem(t *testing.T, opts ...faultDisputeConfigOpts) (*op_
+		precompile := common.BytesToAddress([]byte{0x0a})
+		opts.Gas = 100_000
+		opts.ToAddr = &precompile
+		opts.Data = common.Hex2Bytes("01e798154708fe7789429634053cbf9f99b619f9f084048927333fce637f549b564c0a11a0f704f4fc3e8acfe0f8245f0ad1347b378fbf96e206da11a5d3630624d25032e67a7e6a4910df5834b8fe70e6bcfeeac0352434196bdf4b2485d5a18f59a8d2a1a625a17f3fea0fe5eb8c896db3764f3185481bc22f91b4aaffcca25f26936857bc3a7c2539ea8ec3a952b7873033e038326e87ed3e1276fd140253fa08e9fc25fb2d9a98527fc22a2c9612fbeafdad446cbc7 ...
```
```suggestion
		opts.Data = common.FromHex("01e798154708fe7789429634053cbf9f99b619f9f084048927333fce637f549b564c0a11a0f704f4fc3e8acfe0f8245f0ad1347b378fbf96e206da11a5d3630624d25032e67a7e6a4910df5834b8fe70e6bcfeeac0352434196bdf4b2485d5a18f59a8d2a1a625a17f3fea0fe5eb8c896db3764f3185481bc22f91b4aaffcca25f26936857bc3a7c2539ea8ec3a952b7873033e038326e87ed3e1276fd140253fa08e9fc25fb2d9a98527fc22a2c9612fbeafdad446cbc7bcdbdcd780af2c16a")
```

ajsutton reviewed (approved):
LGTM. Most just need to make semgrep happy by using FromHex instead of Hex2Bytes.

Inphi committed:
fault-proofs: increase max game depth

This is a temporary solution to avoid flakes in e2e tests that use the kzg precompile.
The current execution trace max step of 2**31 steps isn't sufficient.

Inphi commented on packages/contracts-bedrock/deploy-config/devnetL1-template.json:
```diff
@@ -50,7 +50,7 @@
   "recommendedProtocolVersion": "0x0000000000000000000000000000000000000000000000000000000000000000",
   "faultGameAbsolutePrestate": "0x03c7ae758795765c6664a5d39bf63841c71ff191e9189522bad8ebff5d4eca98",
-  "faultGameMaxDepth": 46,
+  "faultGameMaxDepth": 50,
```
@clabby The newly added e2e tests sometimes flakes due to insufficient execution depth. I'm bumping this up just a bit to avoid those flakes.

clabby replied:
ack - no worries.
